proc/proc.go

```diff
@@ -359,10 +359,27 @@ func (dbp *Process) setChanRecvBreakpoints() (int, error) {
 
 // Resume process.
 func (dbp *Process) Continue() error {
+	// all threads stopped over a breakpoint are made to step over it
 	for _, thread := range dbp.Threads {
-		err := thread.Continue()
-		if err != nil {
-			return fmt.Errorf("could not continue thread %d %s", thread.Id, err)
+		if thread.CurrentBreakpoint != nil {
+			if err := thread.Step(); err != nil {
+				return err
+			}
+			if err := thread.SetCurrentBreakpoint(thread.CurrentBreakpoint); err != nil {
```

Comment:
On second thought, maybe move this to `dbp.run`, that's the central function that wraps any other function that continues execution.